Subject: [RFC] hostap: Allow linking with libnl-3.
Date: Tue, 15 Nov 2011 14:30:04 -0000
From: Ben Greear <greearb@candelatech.com>

I needed this patch to compile against the latest
libnl code.  I added this to my config file:

CONFIG_LIBNL32=y

Signed-hostap: Ben Greear <greearb@candelatech.com>
rediffed against 0.7.3 by Stefan Lippers-Hollmann <s.l-h@gmx.de>

---
:100644 100644 5caeec5... 0cc81f9... M  src/drivers/drivers.mak
 src/drivers/drivers.mak |   21 ++++++++++++++-------
 2 files changed, 29 insertions(+), 14 deletions(-)

--- a/src/drivers/drivers.mak
+++ b/src/drivers/drivers.mak
@@ -31,11 +31,18 @@ NEED_SME=y
 NEED_AP_MLME=y
 NEED_NETLINK=y
 NEED_LINUX_IOCTL=y
-DRV_LIBS += -lnl
+ifdef CONFIG_LIBNL32
+  DRV_LIBS += -lnl-3
+  DRV_LIBS += -lnl-genl-3
+  DRV_CFLAGS += -DCONFIG_LIBNL20
+  DRV_CFLAGS += -I/usr/include/libnl3/
+else
+  DRV_LIBS += -lnl
 
-ifdef CONFIG_LIBNL20
-DRV_LIBS += -lnl-genl
-DRV_CFLAGS += -DCONFIG_LIBNL20
+  ifdef CONFIG_LIBNL20
+    DRV_LIBS += -lnl-genl
+    DRV_CFLAGS += -DCONFIG_LIBNL20
+  endif
 endif
 endif
 
